src/main/java/net/masterthought/cucumber/json/Match.java

```diff
@@ -1,13 +1,23 @@
 package net.masterthought.cucumber.json;
 
+import com.google.gson.JsonElement;
+
 public class Match {
 
     // Start: attributes from JSON file report
-    private final String location = null;
+    private final JsonElement location = null;
     // End: attributes from JSON file report
 
     public String getLocation() {
-        return location;
+        if (location.isJsonPrimitive() && location.getAsJsonPrimitive().isString()) {
+            return location.getAsString();
+        } else {
+            StringBuilder sb = new StringBuilder();
+            sb.append(location.getAsJsonObject().get("filepath").getAsJsonObject().get("filename").getAsString());
+            sb.append(":");
+            sb.append(location.getAsJsonObject().get("lines").getAsJsonObject().get("data").getAsJsonArray().get(0).getAsString());
```

Comment:
I'llll try to write it in a way to behave as same as location's 'to_s' method from ruby 
(https://github.com/cucumber/cucumber-ruby-core/blob/master/spec/cucumber/core/ast/location_spec.rb)